Tests/TrackerRadarKitTests/ContentBlockerRulesBuilderTests.swift

```diff
@@ -43,4 +47,37 @@ class ContentBlockerRulesBuilderTests: XCTestCase {
         }
     }
 
+    func testLoadingRulesIsDeterministic() {
+        let firstGeneration = ContentBlockerRulesBuilder(trackerData: trackerData).buildRules(
+            withExceptions: [],
+            andTemporaryUnprotectedDomains: []
+        ).map { $0.trigger.urlFilter }
+
+        let secondGeneration = ContentBlockerRulesBuilder(trackerData: trackerData).buildRules(
+            withExceptions: [],
+            andTemporaryUnprotectedDomains: []
+        ).map { $0.trigger.urlFilter }
+
+        // The data set is large enough that comparing these is slow, so check batches at the start and end.
+        XCTAssertEqual(firstGeneration.prefix(upTo: 1000), secondGeneration.prefix(upTo: 1000))
+        XCTAssertEqual(firstGeneration.suffix(1000), secondGeneration.suffix(1000))
```

Comment:
Open to a better test case here; comparing both arrays in their entirety is extremely slow, and I just needed something quick & dirty that would help isolate the bugs.